From 36ec11ccf3275c6ab9156625d322def9c324e090 Mon Sep 17 00:00:00 2001
From: mozman <me@mozman.at>
Date: Wed, 6 Nov 2019 05:24:45 +0100
Subject: [PATCH] ignore svg tag at pretty print testing

Order of svg attribute of the svg tag has changed in Python 3.8,
ignoring this tag is the easiest way to solve this problem.
---
 tests/test_pretty_xml.py | 3 ++-
 tox.ini                  | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/tests/test_pretty_xml.py b/tests/test_pretty_xml.py
index c004626..14a826e 100644
--- a/tests/test_pretty_xml.py
+++ b/tests/test_pretty_xml.py
@@ -16,7 +16,8 @@ class TestPrettyXML(unittest.TestCase):
     def test_pretty_print(self):
         result = pretty_xml(XML_UGLY).split('\n')
         expect = XML_PRETTY.split('\n')
-        for e, r in zip(expect, result):
+        # skip svg tag, order of attributes changes often by new Python releases
+        for e, r in zip(expect[1:], result[1:]):
             self.assertEqual(e, r)
 
     def test_empty_string(self):
diff --git a/tox.ini b/tox.ini
index 55de9ce..fe450be 100644
--- a/tox.ini
+++ b/tox.ini
@@ -4,7 +4,7 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py36, py37, pypy
+envlist = py36, py37, py38, pypy
 
 [testenv]
 commands = pytest tests
